### PR TITLE
Generate documentation content only

### PR DESCRIPTION
**Pull request by tmattio**
Now that the documention layout and package metadata can be generated dynamically, this PR updates voodoo to only output the required information:

- The raw documentation content (without the layout, or any CSS)
- A table of content in JSON format

This allows us to serve the documentation very fast by not having to generate the content dynamically while keeping the layout and styling dynamic and on the frontend side.

In addition to the simplification of the HTML generator and the output of the JSON file, this PR:

- Removes `voodoo-web` and `voodoo-client`, as they have been migrated to the server
- Unvendor omd and update the Markdown module to the latest AST
- Update the `voodoo-gen` cli to remove the subcommands as we don't need the packages.json and other files anymore

### CODE REVIEW

File: src/voodoo-prep/main.ml [hunk 1]
```diff
@@ -32,7 +32,9 @@ let process_package : Fpath.t -> Package.t -> Fpath.t list -> unit =
     let no_ext = Fpath.rem_ext filename in
     let has_hyphen = String.contains (Fpath.to_string filename) '-' in
     let is_module =
-      process_ocaml_artefacts && List.mem ext [ ".cmt"; ".cmti"; ".cmi" ] && not has_hyphen
+      process_ocaml_artefacts
```
Comment (TheLortex) [inline src/voodoo-prep/main.ml:35]: This change will unfortunately cause a full rebuild of the preparation step. It would be nice to avoid it.

**Review by TheLortex — CHANGES_REQUESTED** [review]
Thank you for this PR, ocaml-docs-ci is being updated to reflect the changes. 

Now it'd be nice to only test the changes affecting voodoo-gen, and keep voodoo-do and voodoo-prep identical so that the CI doesn't have to re-build everything. 

Can you move the voodoo-do / voodoo-prep changes in a separate PR ?

**Comment by TheLortex** [comment]
I've opened https://github.com/tmattio/voodoo/pull/1 to fix the CI

File: src/voodoo-prep/main.ml (hunk 1 above)
Comment (tmattio) [reply src/voodoo-prep/main.ml]: Thanks, I've reverted the changes on voodoo-do and voodoo-prep in d2b485e. We can revert this commit when we've confirmed that the pipeline runs fine, so let's remember to not squash the PR.

**Comment by TheLortex** [comment]
Ok thanks, this looks good. When the updated `ocaml-docs-ci` is live we can merge the PR

**Pull request merged**